test/serializers/links_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class LinksTest < Minitest::Test
+      def setup
+        ActionController::Base.cache_store.clear
+        @blog = Blog.new(id: 1,
+                         name: 'AMS Hints',
+                         writer: Author.new(id: 2, name: "Steve"),
+                         articles: [Post.new(id: 3, title: "AMS")])
+      end
+
+      def test_links_is_present_with_root
+        serializer = AlternateBlogSerializer.new(@blog, :links => {:self => "/blogs/1"})
+        adapter = ActiveModel::Serializer::Adapter::JsonApi.new(serializer)
+        expected = {
+          data: {
+            id: "1",
+            type: "blogs",
+            attributes: {
+              title: "AMS Hints"
+            }
+          },
+          links: {
+            self: "/blogs/1"
+          }
+        }
+        assert_equal expected, adapter.as_json
+      end
+
+      def test_links_is_not_present_when_not_declared
+        serializer = AlternateBlogSerializer.new(@blog)
+        adapter = ActiveModel::Serializer::Adapter::JsonApi.new(serializer)
```

Comment:
Interestingly, I think this may conflict with [#1041 which will generate links based on being passed  `pagination: true`](https://github.com/rails-api/active_model_serializers/pull/1041/files#diff-0ade93904391e3288fa9089f3680d913R5) (i.e. that it doesn't require a links option to generate links).  Food for thought.